client/internal/wssender.go

```diff
@@ -62,10 +73,43 @@ func (s *WSSender) StoppingErr() error {
 	return s.err
 }
 
+// SetHeartbeatInterval sets the heartbeat interval and triggers timer reset.
+func (s *WSSender) SetHeartbeatInterval(d time.Duration) {
+	s.heartbeatIntervalSeconds.Store(int64(d.Seconds()))
+	select {
+	case s.heartbeatIntervalUpdated <- struct{}{}:
+	default:
+	}
+}
+
+func (s *WSSender) shouldSendHeartbeat() <-chan time.Time {
+	t := s.heartbeatTimer
+
+	if !t.Stop() {
+		select {
+		case <-t.C:
+		default:
+		}
+	}
+
+	if d := time.Duration(s.heartbeatIntervalSeconds.Load()) * time.Second; d != 0 {
+		t.Reset(d)
+		return t.C
+	}
+
+	// Heartbeat interval is set to Zero, disable heartbeat.
+	return nil
+}
+
 func (s *WSSender) run(ctx context.Context) {
 out:
 	for {
 		select {
+		case <-s.shouldSendHeartbeat():
+			s.NextMessage().Update(func(msg *protobufs.AgentToServer) {})
```

Comment:
Is this guaranteed to include `instance_uid` field in the next message like the spec requires for heartbeats?

Reply:
`ClientCommon.PrepareStart()` will set `instance_uid` at the start and `SenderCommon` will include the `instance_uid` field in the message. This works in the same way as HTTP Polling.